### ISSUE TITLE
Kingdom Hearts 358/2 Days UNDUB

### ISSUE THREAD

**Issue by Bolleke09** [DS-Homebrew/nds-bootstrap#1207]
Regular Kingdom Hearts 358/2 Days ROM works but the UNDUB found on Cdromance doesn't start the first cutscene. I remember this working on an older version but not on current nightly or stable. 

Test on DSi using hiyacfw and twilightmenu

**Comment by RocketRobz** [comment]
1. Make an xdelta patch of the undub by using the clean ROM as a base.
2. AP-patch the clean ROM using DS-Scene ROM tool.
3. Patch the AP-patched ROM with the undub patch.

**Comment by Bolleke09** [comment]
I'm gonna be honest with you, I am no expert and I can apply an existing
patch but I don't know how to make the UNDUB patch since I can't find it
online. I can only find pre patched ROM for UNDUB.

**Comment by RocketRobz** [comment]
The patch can be made by using the same program used for applying patches, depending on which program you use.

**Comment by Bolleke09** [comment]
But I didn't patch the ROM to begin with I downloaded pre patched 😅, I
also don't think there is a patch online to apply the UNDUB to a clean ROM.
Just a pre patched ROM.

Is there anyway you can help me with this?

**Comment by RocketRobz** [comment]
I'm saying to create the undub patch from the pre-patched undub ROM, so that you can apply the undub patch _you made_ to the AP-patched ROM.
How do you not get that?

**Comment by Bolleke09** [comment]
How do I make the UNDUB patch from the pre patched ROM? Is there a tool for
that?

**Comment by RocketRobz** [comment]
Yes. As said before:
> The patch can be made by using the same program used for applying patches, depending on which program you use.

**Comment by Bolleke09** [comment]
Okay I have never applied any patch before to be honest. The other "clean"
ROM doesn't say it uses an AP patch from the twilight interface when I boot
it. So I assume it AP patched. What tool would you recommend? I use to
create the UNDUB patch to apply to the other ROM?

**Comment by RocketRobz** [comment]
You can use the xDelta patcher (that's what it's called, I think). The UI version is callled xDeltaUI.

**Comment by Bolleke09** [comment]
I tried doing this using several other 358/2 days clean roms. But i am guessing the issue is part of what goes into the patch because when i then use the patch i made using method above the rom doesn't start the first cutscene again just like the original UNDUB rom. I added a .zip with the xdelta patch maybe you can see what's wrong with it. Thank you for all the help i am a total noob and really want to play this version :D
https://easyupload.io/atv4wr

**Comment by remlei** [comment]
yeah every UNDUB games (basically anything that is modified the ROM) may not work at all on nds-bootstrap. I think a simple replacing one file in the ROM can break it (though it works fine on emulators and other flashcarts). Its probably the same issue as I filed here before https://github.com/DS-Homebrew/nds-bootstrap/issues/1192

basically any kinds of modification on the ROM (other than AP patches) can make the ROM incompatible to nds-boostrap. So no fan translation roms and no undub roms is probably compatible with nds-bootstrap.

**Comment by Bolleke09** [comment]
Well honestly every UNDUB I tried and fantranslated game works fine just
not this one 😅

**Comment by Kaisaan** [comment]
I did some searching and I found this [GBAtemp thread](https://gbatemp.net/threads/kingdom-hearts-358-2-days-u-fix.181746/) that apparently fixes this via hex editing the ROM. Specifically changing address 0x010E5DC from 0x0C to 0x37. I'm not able to test this right now but can you verify if this works? The replies only mention this fix working for flashcards and not nds-bootstrap.

**Comment by DarthMotzkus** [comment]
Hello, same problem here with Twilight++ and R4 Gold too. 

@Kaisaan this adress only exists on the non-undub version of this rom. Tried to make an delta patch as @RocketRobz says with the undub rom and then aplied to a ap fixed stock rom, both EU or USA (the undub seems US as DS-SCENE app says thou) and no juice. Wating for you guys to have a look into it. 
Thanks.

**Comment by DarthMotzkus** [comment]
Well i managed to get it to work in Twilight++, simple made the UNDUB process with an previously US rom AP Fixed.
I followed this video, work for US rom too (in the video was done with EU rom wich i cannot play via Twilight but Ys menu plays it normally via R4).

PM me in twitter for any "help", @DarthMotzkus there too.

**Comment by Wolf-lbh** [comment]
> Well i managed to get it to work in Twilight++, simple made the UNDUB process with an previously US rom AP Fixed. I followed this video (https://www.youtube.com/watch?v=YPEh62wz44A), work for US rom too (in the video was done with EU rom wich i cannot play via Twilight but Ys menu plays it normally via R4).
> 
> PM me in twitter for any "help", @DarthMotzkus there too.

Darth that video is now private. I'm having similar problems with several ds games and would really appreciate any help you could give me because the ap fix needs to be applied first but then xdelta gives me an error because that changes the checksum of the rom and it won't let me apply the undub patches to an ap patched rom and I'm getting really frustrated. Also Twitter removed private messaging for some reason. Do you have a discord @? The games I'm having trouble with ap fixes + patching are:
FF Crystal Chronicles - Echoes of Time
Inazuma Eleven
Kingdom Hearts Re:Coded
Kingdom Hearts 385/2 Days
Zelda - Spirit Tracks (The patch I'm trying to apply is a patch for d-pad controls)

**Comment by Wolf-lbh** [comment]
Update: I figured out how to get everything working!
So the issue I was having was that the ap patch needs to be applied first for stability.
But the second patch would then give an error and say the checksum had failed because the base rom was edited by the ap patch.

Turns out I just had to switch the xdelta patching program I was using to "DeltaPatcher" which you can click the setting wheel and uncheck "Checksum Validation" and it will let you apply xdelta patches to modified roms.

Note: You should leave this setting on by default because any other time you get a checksum warning it means you are using the wrong rom or it is corrupted/modified in some way that will either cause the game to not boot or corrupt when you play it.
